r-package/grf/vignettes/rate_cv.Rmd

```diff
@@ -0,0 +1,438 @@
+  )
+}))
+paste("Rejection rate:", mean(res) * 100, "%")
+#> [1] "Rejection rate: 5.4 %"
+```
+
+Now that we have a valid test for treatment heterogeneity using several folds, let's try it out on a simulated example and compare its performance with simple sample splitting. Below we generate data from synthetic observational data (Setup A in Nie & Wager (2021)) where treatment effect heterogeneity is hard to detect. In this case, we want our test to reject $H_0$ of no heterogeneity.
+
+```{r, eval=FALSE}
+res = t(replicate(500, {
+  # Simulate synthetic data with "weak" treatment heterogeneity
+  n = 2500
+  p = 5
+  data = generate_causal_data(n, p, sigma.tau = 0.15, sigma.noise = 1, dgp = "nw1")
+  X = data$X
+  Y = data$Y
+  W = data$W
+
+  pval.sequential = rate_sequential(X, Y, W)
+
+  # Compare with a single train/test split
+  train = sample(n, n / 2)
+  test = -train
+  cf.train = causal_forest(X[train,], Y[train], W[train])
+  cf.test = causal_forest(X[test,], Y[test], W[test])
+  rate = rank_average_treatment_effect(cf.test, predict(cf.train, X[test, ])$predictions)
+  tstat.split = rate$estimate / rate$std.err
+  pval.split = 2 * pnorm(-abs(tstat.split))
+
+  c(
+    reject.sequential = pval.sequential <= 0.05,
+    reject.split = pval.split <= 0.05
+  )
+}))
+colMeans(res) * 100
+#> reject.sequential    reject.split
+#> 56.6                 47.0
+```
+
+Reassuringly, in this example, cross-fold estimation is able to improve power to detect treatment heterogeneity over a single train/test split.
+
+The sequential estimation construction in this section is motivated by a martingale central limit theorem by Luedtke & van der Laan (2016), and as discussed in Wager (2024) can deliver powerful tests for the presence of treatment heterogeneity.
+
+## Heuristic: out-of-bag estimation with one-sided tests
+The fundamental challenge with estimating CATEs and evaluating the presence of heterogeneity using the same data is overfitting: we're simply constrained by the fact that we can't simply estimate and evaluate something using the same data set. Random forests come with a neat algorithmic workaround for this issue: out-of-bag (OOB) estimation. With this strategy, we can train a forest using all our data and obtain predictions for the same data. In terms of breaking dependence between training and evaluation, out-of-bag estimation is not as strong as sample splitting, but it can serve as a very useful middle ground.
+Performing traditional two-sided tests with OOB-estimated RATEs is problematic as estimates are going to be biased downwards. However, if we are willing to commit to a one-sided test, then we can use OOB estimates and still be relatively confident we are maintaining nominal error guarantees (we caution that this is a *heuristic* not backed by a formal central-limit result). In the example below we compare a two-sided test for RATE $\neq 0$ with a one-sided test for RATE $>0$ at a 5% level.
```

Comment:
There should be more discussion here about how basic OOB is just the leave-one-out version of naive cross-fold estimation, which we already found doesn't work.

The fix of using one sided tests could also be used with usual cross-fold approach.

So there are really two ideas here: One sided tests as a heuristic to fix the naive aggregation, and OOB as a computational speed up with forests.